src/Stateless/StateConfiguration.Async.cs

```diff
@@ -24,8 +24,8 @@ public StateConfiguration InternalTransitionAsyncIf(TTrigger trigger, Func<bool>
             {
                 if (entryAction == null) throw new ArgumentNullException(nameof(entryAction));
 
-                _representation.AddTriggerBehaviour(new InternalTriggerBehaviour(trigger, guard));
-                _representation.AddInternalAction(trigger, (t, args) => entryAction(t));
+                Representation.AddTriggerBehaviour(new InternalTriggerBehaviour(trigger, guard));
```

Comment:
Did you make Representation public? If not, could we keep this as _representation?